internal/databases/mysql/binlog_list_handler.go

```diff
@@ -0,0 +1,146 @@
+package mysql
+
+import (
+	"fmt"
+	"path/filepath"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/wal-g/tracelog"
+	"github.com/wal-g/wal-g/pkg/storages/storage"
+	"github.com/wal-g/wal-g/utility"
+)
+
+type BinlogInfo struct {
+	Name         string    `json:"name"`
+	Size         int64     `json:"size,omitempty"`
+	LastModified time.Time `json:"last_modified,omitempty"`
+}
+
+type TimeFilter struct {
+	Since time.Time
+	Until time.Time
+}
+
+func (tf *TimeFilter) IsValid() bool {
+	if tf.Since.IsZero() || tf.Until.IsZero() {
+		return true
+	}
+	return tf.Since.Before(tf.Until) || tf.Since.Equal(tf.Until)
+}
+
+func (tf *TimeFilter) Matches(t time.Time) bool {
+	if !tf.Since.IsZero() && t.Before(tf.Since) {
+		return false
+	}
+	if !tf.Until.IsZero() && t.After(tf.Until) {
+		return false
+	}
+	return true
+}
+
+func HandleBinlogList(folder storage.Folder, since, until string) {
+	binlogFolder := folder.GetSubFolder(BinlogPath)
+
+	binlogFiles, _, err := binlogFolder.ListFolder()
+	if err != nil {
+		tracelog.ErrorLogger.Printf("Failed to list binlog folder: %v", err)
+		tracelog.ErrorLogger.FatalOnError(err)
```

Comment:
NIT: you can combine 2 lines into one with `tracelog.ErrorLogger.FatalfOnError()`